pycbc/workflow/jobsetup.py

```diff
@@ -1166,19 +1356,92 @@ class LalappsInspinjExecutable(Executable):
     """
     The class used to create jobs for the lalapps_inspinj Executable.
     """
-    current_retention_level = Executable.FINAL_RESULT
-    def create_node(self, segment):
+    current_retention_level = Executable.CRITICAL
+    def __init__(self, cp, exe_name, universe=None, ifos=None, out_dir=None,
+                 tags=[]):
+        Executable.__init__(self, cp, exe_name, universe, ifos, out_dir,
+                            tags=tags)
+        self.cp = cp
+        self.tags = tags
+
+    def create_node(self, segment, exttrig_file=None, tags=[]):
         node = Node(self)
 
-        if self.get_opt('write-compress') is not None:
-            ext = '.xml.gz'
+        curr_tags = self.tags + tags
+        # This allows the desired number of injections to be given explicitly
+        # in the config file. Used for coh_PTF as segment length is unknown
+        # before run time.
+        if 'coh_PTF_inspiral' in self.cp.get('executables', 'inspiral'):
+            num_injs = int(self.cp.get_opt_tags('workflow-injections',
+                                                'num-injs', curr_tags))
+            inj_tspace = float(segment[1] - segment[0]) / num_injs
+            node.add_opt('--time-interval', inj_tspace)
+            node.add_opt('--time-step', inj_tspace)
+            
+            if self.get_opt('l-distr') == 'exttrig':
+                if exttrig_file is None:
+                    raise ValueError("Must supply an external trigger file if"
+                                     " option 'l-distr' is set to 'exttrig'")
+                node.add_opt('--exttrig-file', '%s' % exttrig_file.storage_path)
+            
+            node.new_output_file_opt(segment, '.xml', '--output',
+                                     store_file=self.retain_files)
         else:
-            ext = '.xml'
+            node.new_output_file_opt(segment, '.xml', '--output',
+                                     store_file=self.retain_files)
+        
+        node.add_opt('--gps-start-time', segment[0])
+        node.add_opt('--gps-end-time', segment[1])
```

Comment:
Reverted in https://github.com/a-r-williamson/pycbc/commit/e2910dc7c51949b3e8cc8ab28d6a2e75226c3945